lib/data/moor_database.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:moor/moor.dart';
+
+part 'moor_database.g.dart';
+
+@DataClassName("CalendarItem")
+class CalendarItems extends Table {
+  //autoincrement automatically sets id as the primary key
+  IntColumn get id => integer().autoIncrement()();
+
+  TextColumn get title => text().withLength(min: 1)();
+
+  DateTimeColumn get startTime => dateTime()();
+
+  DateTimeColumn get endTime => dateTime()();
+}
+
+//Table for Todo Items
+@DataClassName("TodoItem")
+class TodoItems extends Table {
+  IntColumn get id => integer().autoIncrement()();
+
+  TextColumn get title => text().withLength(min: 1)();
+
+  //Duration is measured in seconds (since we are sticking mainly to shorter tasks)
+  IntColumn get duration => integer()();
+
+  DateTimeColumn get dateAdded => dateTime()();
+}
+
+@UseMoor(tables: [CalendarItems, TodoItems])
+class LocalDatabase extends _$LocalDatabase {
+  LocalDatabase(QueryExecutor queryExecutor) : super(queryExecutor);
+
+  /*
+  The "version" of the db. Should be incremented whenever changing/adding
+  a table definition.
+  */
+  @override
+  int get schemaVersion => 1;
+}
+
+//Data Access Object for TodoItems table
+// - Where all queries for the TodoItems table will go
+@UseDao(tables: [TodoItems])
+class TodoItemDao extends DatabaseAccessor<LocalDatabase>
+    with _$TodoItemDaoMixin {
+  final LocalDatabase db;
+
+  //This constructor allows local db to create an instance of this DAO
+  TodoItemDao(this.db) : super(db);
+
+  Future<List<TodoItem>> getAllTodoItems() => select(todoItems).get();
+
+  //Required since instantiating a Todo Item without id will have a null id
+  // - The id is only autoincremented after it is inserted into the local db
+  // - So to get the id we must insert it into the db & then retrieve it
+  Future<TodoItem> getTodo(String title) =>
+      (select(todoItems)..where((todoItem) => todoItem.title.equals(title)))
+          .getSingle();
+
+  Stream<List<TodoItem>> watchAllTodoItems() => select(todoItems).watch();
+
+  Future<int> insertTodoItem(TodoItem todoItem) =>
+      into(todoItems).insert(todoItem);
+
+  //Replaces the pre-existing TodoItem with the same id as the todoItem argument
+  Future<bool> updateTodoItem(TodoItem todoItem) =>
+      update(todoItems).replace(todoItem);
+
+  //Deleting based on title of TodoItem
+  // - Might not be the best implementation
```

Comment:
This should be okay. For deleting, we can have the user press on the Todo item and then press some delete button. Then we can call deleteTodoItem on the title variable of that Todo Item.